### PR TITLE
2.0.0

### LINKED ISSUE #642: 2.0.0 Wishlist
@chriddyp @Kully @cldougl 

I haven't had many cycles to play around with the python api library lately, but recently got in a little deep for https://github.com/plotly/plotly.py/pull/640, where we've now got *fully-unified* api modules.

I think enough shifted (mostly additions, but the error handling is definitely not backwards compat) that we should open up a discussion on non-minor changes that we've been looking to add to the plotly module. Here's a list of topics:

- [x] API request/response unification (https://github.com/plotly/plotly.py/pull/651)
- [ ] Fix or drop support for matplotlib
- [x] Add `numpy` as an official dep. and use it to speed up the library. (#667)
- [x]  Related to above, clean up how we handle optional imports (https://github.com/plotly/plotly.py/pull/317)
- [x] Stop auto-loading graph reference and just use packaged graph reference (https://github.com/plotly/plotly.py/pull/614)**
- [x] ~~Add transforms to graph objects (https://github.com/plotly/plotly.py/issues/587)~~ holding off on this until the transforms spec settles in plotly.js.
- [x] Add frames for graph objects (https://github.com/plotly/plotly.py/issues/604)
- [x] Add a sign_in validation #525 (#668)
- [x] ~~Fix up streaming? (this might be more work than it's worth at this point https://github.com/plotly/plotly.py/pull/253)~~ No time for this atm.
- [x] Some module re-organizing. Most importantly, we should move FigureFactory into it's own sub-package and each Figure-generator should be it's own file. The `tools.py` module is enormous and is a constant point of pain for circular imports (https://github.com/plotly/plotly.py/pull/652).

**In fact, if we *do* do this, we can just add some meta programming to auto-create a module full of all the objects from graph-reference. This way we don't have those dump "Scatter is not defined" errors, etc.

Anything else?

### PR DESCRIPTION
**Pull request by theengineear**
Aight, we're set to merge this. I'm going to ask for a final round of review and then merge and setup a `pre-release` for this before I push anything new up to PyPI.

Closes #642

### CODE REVIEW

**Comment by theengineear** [comment]
@chriddyp @Kully @cldougl is there any reason for us to support `1.X` anymore after we do a major version bump? Bigger projects often support (and continue to update minor versions on) all major versions. (see the `stable/X` branches in [django](https://github.com/django/django) for an example). *I don't see a need for us to do this* and we should always be able to go back in time, check out a previous version and cherry-pick in a change if necessary.

In other words, I'd like to keep our versioning linear if possible, each new version should be a higher number than the last. I suppose we can always change this later by pulling old commits into new branches, etc, etc.

 Again, I'm suggesting that we merge `2.0.0` directly into master and not manage `1.X` and `2.X` in separate branches or something.

**Comment by Kully** [comment]
I feel like it would be pretty annoying to maintain both branches. I recall a lot of organization stuff (FigureFactory) which won't affect the users' experience and more additions which make 2.0.0 sounds better.

**Comment by theengineear** [comment]
>I feel like it would be pretty annoying to maintain both branches.

Yup, it would be. Not supporting multiple, concurrent versions is definitely my vote, but I just want to get consensus on our approach.

**Comment by cldougl** [comment]
+1 for not continuing to support version `1.X`
*We'll have to update the docs as well.

**Comment by theengineear** [comment]
Great! Consensus-enough! Then I'm going to merge `2.0.0` to master today and get it up on PyPI.

**Comment by theengineear** [comment]
Okie dokie. Heres' the state of `2.0.0`:

* ensure that layout gets passed in along to `clientresp`: #666
* require `numpy` in FF methods: #667
* validate on `py.sign_in`: #668

I'll version-bump when all that's through ;)

**Comment by theengineear** [comment]
Aight, final tests are happening now!

@chriddyp @cldougl @Kully , any final reservations? I'll create a pre-release tag for this and may ask for a bit more manual testing after this hits `master`.

**Comment by cldougl** [comment]
no reservations!
and I'm happy to help manual test too

**Comment by theengineear** [comment]
OK, done waiting, I think the next step needs to be manual testing. Merge time!
